app/src/main/res/layout/content_main.xml

```diff
@@ -1,37 +1,41 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     app:layout_behavior="@string/appbar_scrolling_view_behavior"
-    tools:showIn="@layout/activity_main"
-    tools:context=".MainActivity">
+    tools:context=".MainActivity"
+    tools:showIn="@layout/activity_main">
 
     <Button
         android:id="@+id/button4"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="View Activities"
-        tools:layout_editor_absoluteX="227dp"
-        tools:layout_editor_absoluteY="375dp" />
+        android:layout_marginTop="61dp"
+        android:text="@string/view_activities"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/button2"
+        app:layout_constraintTop_toBottomOf="@+id/txtTimeSpent" />
 
     <EditText
         android:id="@+id/txtTimeSpent"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
+        android:layout_marginTop="34dp"
         android:ems="10"
+        android:hint="@string/time_spent_hint"
         android:inputType="textPersonName"
-        android:text="Time spent..."
-        tools:layout_editor_absoluteX="99dp"
-        tools:layout_editor_absoluteY="269dp" />
+        android:text="@string/time_spent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/txtActivity" />
 
     <TextView
         android:id="@+id/textView"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="What activity would you like to log?"
+        android:text="@string/activity_log_text"

```

Comment:
Created string in strings.xml and replaced text with @string. This removed this warning in android studio about using string values.

Reply:
Really good.  Putting visible text in strings.xml is internationalization-friendly.  You can have different versions of strings.xml for each of the languages that you support.